sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1647,4 +1643,300 @@ private[spark] object QueryCompilationErrors {
   def invalidYearMonthIntervalType(startFieldName: String, endFieldName: String): Throwable = {
     new AnalysisException(s"'interval $startFieldName to $endFieldName' is invalid.")
   }
+
+  def queryFromRawFilesIncludeCorruptRecordColumnError(): Throwable = {
+    new AnalysisException(
+      """
+        |Since Spark 2.3, the queries from raw JSON/CSV files are disallowed when the
+        |referenced columns only include the internal corrupt record column
+        |(named _corrupt_record by default). For example:
+        |spark.read.schema(schema).csv(file).filter($\"_corrupt_record\".isNotNull).count()
+        |and spark.read.schema(schema).csv(file).select(\"_corrupt_record\").show().
+        |Instead, you can cache or save the parsed results and then send the same query.
+        |For example, val df = spark.read.schema(schema).csv(file).cache() and then
+        |df.filter($\"_corrupt_record\".isNotNull).count().
+      """.stripMargin('#'))
```

Comment:
why use `'#'` here?

Reply:
Thank you.  `stripMargin` is good enough.